When publishing String columns to DH -- IndexOutOfBoundsException

<!-- Please read our Code of Conduct: https://github.com/deephaven/deephaven-core/blob/main/CODE_OF_CONDUCT.md -->
<!-- Please search existing issues to avoid creating duplicates. -->

**Description**

When appending a new table with a String column to an InputTable I sometimes generate an `IndexOutOfBoundsException`.


**Steps to reproduce**

1. Assign `var table = ColumnHeader.of("test", String.class)`
2. Use a FlightSession to execute `InMemoryAppendOnlyInputTable.of(TableHeader.of(ColumnHeader.of("test", String.class)))` and save the handle to `tableHandle`
3. `var newRows = table.start(1000);`
4. In a loop append random length of strings via `newRows.row("...randomStringHere...")`
5. Use `flightSession.addToInputTable(tableHandle, newRows.newTable(), bufferAllocator)`


**Expected results**

The table chunk is appended to the previously created DH input table. 

**Actual results**

```
...
Caused by: java.lang.IndexOutOfBoundsException: index: 32768, length: 16 (expected: range(0, 32768))
	at org.apache.arrow.memory.ArrowBuf.checkIndex(ArrowBuf.java:701)
	at org.apache.arrow.memory.ArrowBuf.setBytes(ArrowBuf.java:765)
	at org.apache.arrow.vector.BaseVariableWidthVector.setBytes(BaseVariableWidthVector.java:1244)
	at org.apache.arrow.vector.BaseVariableWidthVector.set(BaseVariableWidthVector.java:1025)
	at io.deephaven.client.impl.VectorHelper.fill(VectorHelper.java:103)
	at io.deephaven.client.impl.FieldVectorAdapter.visitStringArray(FieldVectorAdapter.java:179)
	at io.deephaven.client.impl.FieldVectorAdapter$1.visit(FieldVectorAdapter.java:92)
	at io.deephaven.qst.type.StringType.walk(StringType.java:27)
	at io.deephaven.client.impl.FieldVectorAdapter.visit(FieldVectorAdapter.java:89)
	at io.deephaven.qst.array.GenericArray.walk(GenericArray.java:55)
	at io.deephaven.client.impl.FieldVectorAdapter.of(FieldVectorAdapter.java:65)
	at io.deephaven.client.impl.FieldVectorAdapter.of(FieldVectorAdapter.java:53)
	at io.deephaven.client.impl.VectorSchemaRootAdapter.of(VectorSchemaRootAdapter.java:33)
	at io.deephaven.client.impl.FlightClientHelper.put(FlightClientHelper.java:34)
Caused by: java.lang.reflect.InvocationTargetException: null

	at io.deephaven.client.impl.FlightClientHelper.put(FlightClientHelper.java:25)
	at io.deephaven.client.impl.FlightSession.put(FlightSession.java:209)
Caused by: java.lang.IndexOutOfBoundsException: index: 32768, length: 16 (expected: range(0, 32768))
...
```

**Additional details and attachments**

I've traced this down to [this line](https://github.com/deephaven/deephaven-core/blob/a2db33d190d01e3ceaed07def3e47aa62fb9b9e1/java-client/flight/src/main/java/io/deephaven/client/impl/VectorHelper.java#L103) -- basically VarCharVector which we're using here allocates a standard size chunk (32k) and we keep putting bytes into it w/o bounds checking and reallocating, because the above is using `set`. A simple solution (tested locally) -- at the cost of some performance and memory management in the cases where 32k is not enough for a column chunk -- would be to replace this call with `setSafe` instead: [link](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/BaseVariableWidthVector.html#setSafe-int-byte:A-)


**Versions**
 - Deephaven: 0.25.1+

--- a/java-client/flight/src/main/java/io/deephaven/client/impl/VectorHelper.java
+++ b/java-client/flight/src/main/java/io/deephaven/client/impl/VectorHelper.java
@@ -100,7 +100,7 @@ public class VectorHelper {
             if (value == null) {
                 vector.setNull(i);
             } else {
-                vector.set(i, value.getBytes(StandardCharsets.UTF_8));
+                vector.setSafe(i, value.getBytes(StandardCharsets.UTF_8));
             }
             ++i;
         }

